### PR TITLE
QMCPACK Update July 2019

### PR DESCRIPTION
**Pull request by naromero77**
Incremental Update to QMCPACK Spack package, includes:
- Support for QE 6.4.1 converter
- Support for QMCPACK 3.8.0

@adamjstewart @scheibelp

### CODE REVIEW

File: var/spack/repos/builtin/packages/qmcpack/package.py [hunk 1]
```diff
@@ -141,17 +135,15 @@ class Qmcpack(CMakePackage, CudaPackage):
     depends_on('py-matplotlib', when='+gui', type='run')
 
     # B-spline basis calculation require a patched version of
-    # Quantum Espresso 6.4 (see QMCPACK manual)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qmcpack/package.py]: Is it possible to keep 6.4 in addition to 6.4.1? We try not to remove old versions of packages for reproducibility.

Comment (naromero77) [reply var/spack/repos/builtin/packages/qmcpack/package.py]: You have raised this issue in the past, but we don't maintain old QE converters and have a tendency to make the old ones disappear :)

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/qmcpack/package.py]: Thanks, you'll prob have to remind me each time, sorry for being a broken record 😛

**Comment by naromero77** [comment]
@adamjstewart Can this be merged in? 

I don’t understand why the codecov test is failing.

**Pull request merged**